projects/plugins/jetpack/modules/widget-visibility/widget-conditions/widget-conditions.js

```diff
@@ -311,3 +311,328 @@ jQuery( function ( $ ) {
+			},
+		} );
+	}
+	return settings;
+}
+
+wp.hooks.addFilter( 'blocks.registerBlockType', 'widget/visibility', addVisibilityAttribute );
+
+const VisibilityRule = props => {
+	console.log( { widget_conditions_parent_pages, widget_conditions_data } );
+	const { i, rule, onDelete, setMajor, setMinor } = props;
+	const { createElement } = wp.element;
+	const { __ } = wp.i18n;
+	const { Button, SelectControl } = wp.components;
+
+	let majorOptions = [
+		{ label: __( '-- Select --', 'jetpack' ), value: '' },
+		{ label: __( 'Category', 'jetpack' ), value: 'category' },
+		{ label: __( 'Author', 'jetpack' ), value: 'author' },
+		{ label: __( 'User', 'jetpack' ), value: 'loggedin' }, // TURN OFF FOR WPCOM
+		{ label: __( 'Role', 'jetpack' ), value: 'role' }, // TURN OFF FOR WPCOM
+		{ label: __( 'Tag', 'jetpack' ), value: 'tag' },
+		{ label: __( 'Date', 'jetpack' ), value: 'date' },
+		{ label: __( 'Page', 'jetpack' ), value: 'page' },
+		{ label: __( 'Taxonomy', 'jetpack' ), value: 'taxonomy' }, // ONLY IF A NON-DEFAULT TAXON IS FOUND
+	];
+
+	let minorOptions = [];
+	if ( rule.major in widget_conditions_data ) {
+		minorOptions = widget_conditions_data[ rule.major ].map( ( [ x, y ] ) => ( {
+			value: x,
+			label: y,
+		} ) );
+		// TODO: *Page* has nested arrays
+		// We don't handle these yet
+		// [
+		//     [ "front", "Front page" ],
+		//     [ "posts", "Posts page" ],
+		//     [ "archive", "Archive page" ],
+		//     [ "404", "404 error page" ],
+		//     [ "search", "Search results" ],
+		//     [
+		//         "Post type:",
+		//         [
+		//             [ "post_type-post", "Post" ],
+		//             [ "post_type-page", "Page" ],
+		//             [ "post_type-attachment", "Media" ]
+		//         ]
+		//     ],
+		// ]
+	}
+
+	/*
+	let a = (
+		<div>
+			This is rule number { i }
+			<SelectControl
+				label="Major Rule"
+				value={ rule.major }
+				options={ majorOptions }
+				onChange={ setMajor }
+			/>
+			is
+			{ rule.major && (
+				<SelectControl
+					label="Minor Rule"
+					value={ rule.minor }
+					options={ minorOptions }
+					onChange={ setMinor }
+				/>
+			) }
+			<Button onClick={ onDelete }>{ __( 'Delete' ) }</Button>
+		</div>
+	);
+    */
+
+	return createElement(
+		'div',
+		null,
+		'This is rule number ',
+		i,
+		createElement( SelectControl, {
+			label: 'Major Rule',
+			value: rule.major,
+			options: majorOptions,
+			onChange: setMajor,
+		} ),
+		'is',
+		rule.major &&
+			createElement( SelectControl, {
+				label: 'Minor Rule',
+				value: rule.minor,
+				options: minorOptions,
+				onChange: setMinor,
+			} ),
+		createElement( Button, { onClick: onDelete }, __( 'Delete' ) )
+	);
+};
+
+const RuleSep = props => {
+	const { isAnd } = props;
+	const { createElement } = wp.element;
+	if ( isAnd ) {
+		return createElement( 'div', null, 'and' );
+	}
+	return createElement( 'div', null, 'or' );
+};
+
+const visibilityAdvancedControls = wp.compose.createHigherOrderComponent( BlockEdit => {
```

Comment:
On the classic widgets screen (can be enabled by installing Classic Widgets plugin from core and then going to /wp-admin/widgets.php wp.compose won't be available so this will result in an uncaught TypeError